public/app/plugins/datasource/loki/language_provider.ts

```diff
@@ -129,7 +129,7 @@ export default class LokiLanguageProvider extends LanguageProvider {
     const { wrapperClasses, value, prefix, text } = input;
 
     // Local text properties
-    const empty = value.document.text.length === 0;
+    const empty = value?.document?.text?.length === 0;
```

Comment:
is those really possible to be null? how did it not result in null exceptions? is the type wrong?

Reply:
I think only `value` can be undefined so should be `value?.document.text.length` similar to other places.